scheduler/src/cook/mesos/api.clj

```diff
@@ -1858,6 +1851,40 @@
         json-response (res/content-type "application/json")
         (and json-response body) (assoc :body (streaming-json-encoder body))))))
 
+(def cook-coercer
+  "This coercer adds to compojure-api's default-coercion-matchers by
+  converting keys from snake_case to kebab-case for requests and from
+  kebab-case to snake_case for responses. For example, this allows clients to
+  pass 'max_retries' when submitting a job, even though the job schema has
+  :max-retries (with a dash instead of an underscore). For more information on
+  coercion in compojure-api, see:
+
+    https://github.com/metosin/compojure-api/wiki/Validation-and-coercion"
+  (constantly
+    (-> c-mw/default-coercion-matchers
+      (update :body
+              (fn [their-matchers]
+                (fn [s]
+                  (or (their-matchers s)
+                      (get {;; can't use form->kebab-case because env and label
+                            ;; accept arbitrary kvs
+                            JobRequestMap (partial map-keys ->kebab-case)
+                            Group (partial map-keys ->kebab-case)
+                            HostPlacement (fn [hp]
+                                            (update hp :type keyword))
+                            UpdateRetriesRequest (partial map-keys ->kebab-case)
```

Comment:
The only thing I changed in this function was the addition of this line. I moved the whole function down here because it was being defined before `UpdateRetriesRequest`. Since it's not used earlier in the file, I moved it as far down as possible so that we can avoid moving it again in the future.